backend/server/game_api.js

```diff
@@ -0,0 +1,637 @@
+    if (shipInDb.start_row === shipInDb.end_row) {
+        if ((start_row === shipInDb.start_row && start_col === (shipInDb.start_col - 1)) || (start_row === shipInDb.start_row && start_col === (shipInDb.end_col + 1))) {
+            return 1;
+        }
+
+        if (start_row === (shipInDb.start_row + 1) && (start_col >= (shipInDb.start_col - 1) && start_col <= (shipInDb.end_col + 1))){
+            return 1;
+        }
+
+        if (start_row === (shipInDb.start_row - 1) && (start_col >= (shipInDb.start_col - 1) && start_col <= (shipInDb.end_col + 1))){
+            return 1;
+        }
+
+        if (start_row < shipInDb.start_row && end_row > shipInDb.start_row && (start_col >= (shipInDb.start_col - 1) && start_col <= (shipInDb.end_col + 1))) {
+            return 1;
+        }
+    } else if (shipInDb.start_row !== shipInDb.end_row) {
+        if (start_col === (shipInDb.start_col - 1) && (start_row === shipInDb.start_row || start_row === shipInDb.end_row)) {
+            return 1;
+        }
+
+        if (start_col === (shipInDb.start_col + 1) && (start_row === shipInDb.start_row || start_row === shipInDb.end_row)) {
+            return 1;
+        }
+
+        if (start_row === (shipInDb.start_row - 1) && (start_col >= (shipInDb.start_col - 1) && start_col <= (shipInDb.start_col + 1))) {
+            return 1;
+        }
+
+        if (start_row === (shipInDb.end_row + 1) && (start_col >= (shipInDb.start_col - 1) && start_col <= (shipInDb.start_col + 1))) {
+            return 1;
+        }
+
+        if (start_col < shipInDb.start_col && end_col > shipInDb.start_col && (start_row >= (shipInDb.start_row - 1) && start_row <= (shipInDb.end_row + 1))) {
+            return 1;
+        }
+    }
+    return 0;
+};
+
+
+async function checkShips(req, dimension, ship_size) {
+    const userShips = await pool.query('select size from ships where game_id = $1 and user_id = $2', [req.params.game_id, req.user.id]);
+    const gameShips = shipsAmount[dimension];
+    const placedShips = userShips.rows.map(ship => ship.size);
+
+    let count = 0;
+    for (let i = 0; i < placedShips.length; i++) {
+        if (ship_size === placedShips[i]) {
+            count += 1;
+        }
+    };
+
+    if (count < gameShips[ship_size]) {
+        return true; //'can place'
+    } else {
+        return false; //'cannot place'
+    };
+};
+
+
+async function placeShip(game_id, user_id, ship_size, start_row, start_col, end_row, end_col, res) {
+    await pool.query('insert into ships (game_id, user_id, size, start_row, start_col, end_row, end_col) values ($1, $2, $3, $4, $5, $6, $7)', [game_id, user_id, ship_size, start_row, start_col, end_row, end_col]);
+    return res.status(200).json({msg: `Placed a ship of size ${ship_size}`});
+};
+
+
+gameRouter.post('/games/:game_id/place', async (req, res) => {
+    const { ship_size, start_row, start_col, end_row, end_col } = req.body;
+
+    if (start_row === end_row) {
```

Comment:
move this verification to a function with a meaningful name that returns `true`/`false`